src/edgepi/adc/edgepi_adc.py

```diff
@@ -979,6 +987,136 @@ def set_config(
                                         [None])
         self.__config(**args)
 
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-statements
+    def batch_read_samples_adc1(
+        self,
+        data_rate: ADC1DataRate,
+        analog_in_list: Optional[list[AnalogIn]] = None,
+        differential_pairs: Optional[list[DiffMode]] = None,
+    ) -> list:
+        """
+        This function gets a data rate (to set ADC1 to), and a list of analog in and
+        differential pairs to read from. The differential pairs & analog_in elements may overlap,
+        but usecases are be uncommon.
+
+        Will reset any prior config made to the ADC. Does not work with RTD mode, and may override configs if RTD mode
+        is active.
+
+        This function only supports ADC 1, and changes the conversion mode to PULSE automatically.
+        
+        This function will not cache it's resulting register values. # TODO: change this?
+        
+        TODO: if even more performance is needed, continuous mode might only need to sleep for the first conversion!
+        """
+        analog_in_list     = [] if analog_in_list is None else analog_in_list
+        differential_pairs = [] if differential_pairs is None else differential_pairs
+
+        if analog_in_list == [] and differential_pairs == []:
+            raise ValueError("Both analog_in_list and differential_pairs cannot be empty")
+
+        channel_list = [self.__analog_in_to_adc_in_map.get(analog_in) for analog_in in analog_in_list]
+        if any(True for ch in channel_list if ch is None):
+            raise ValueError(f"Invalid analog_in_list={analog_in_list}")
+
+        # get current register values by doing an SPI read
+        register_values = self.__get_register_map()
+        if len(register_values.values()) < 1:
+            raise ValueError("Number of reg_values must be at least 1")
+
+        # determine the conversion delay using the table from the datasheet (best case is sleep of 0.207ms)
+        filter_mode_op_code = (~ADCProperties.FILTER_MODE.value.mask) & register_values[ADCProperties.FILTER_MODE.value.addx]
+        conversion_delay = expected_initial_time_delay(
+            ADCNum.ADC_1, data_rate.value.op_code, filter_mode_op_code
+        ) / 1000
+
+        # this is the register value of MODE2 that contains the updated data rate
+        mode2_register_value = (
+            register_values[ADCReg.REG_MODE2.value]
+            & data_rate.value.op_mask
+        ) | data_rate.value.op_code
+
+        def create_read_command_tup(is_first: bool, mux_p: CH, mux_n: CH) -> tuple:
```

Comment:
Can this be defined outside `batch_read_samples_adc1`, to shorten the length of this method.